### DISCUSSION TITLE
What exactly does the staleTime parameter is for ?

### DISCUSSION THREAD

**Discussion by dardar4** [TanStack/query#1013] [General] [answered]
I understand that a quest is becoming stale after X amount of time, but what does it saying about that query exactly?

or better yet - what is the best practice in case I want my query to refresh every N minutes ? is it done by the staleTime or the cacheTime?

thanks!

**Comment by denisborovikov** [comment, marked as answer]
Here are the answers what are stale/cache times for https://react-query.tanstack.com/docs/guides/caching

Here's an example how to implement auto refetching/polling https://react-query.tanstack.com/docs/examples/auto-refetching

**Reply by dardar4** [reply]
thanks!

I still not sure what the following sentence means:
"Both instances of the useQuery('todos', fetchTodos) query are unmounted and no longer in use"

when queries are unmounted ?

**Reply by denisborovikov** [reply]
It means when the components with `useQuery` are unmounted, the queries are unmounted as well.
